state_machine/docs/README.md

```diff
@@ -0,0 +1,66 @@
+# State Machine Module
+
+This document outlines the purpose of this module, its components and how they all interact with the other modules.
+
+## Contents <!-- omit in toc -->
+- [State Machine Module](#state-machine-module)
+  - [Overview](#overview)
+  - [Code Structure](#code-structure)
+  - [High Level Architecture](#high-level-architecture)
+    - [FSM primer](#fsm-primer)
+  - [Current State Machine Definition](#current-state-machine-definition)
+
+## Overview
+
+The `StateMachine` module implements a FSM (Finite State Machine) that is responsible for managing the node lifecycle since its internal behaviour can be different depending on certain conditions that are used to determine the current state.
+
+TL;DR: When you are in a state, you can only transition to other states that are allowed by the fsm definition.
+Transitions happen reacting to events.
+
+
+## Code Structure
+
+```bash
+├── docs
+│   ├── CHANGELOG.md              # Changelog
+│   ├── README.md                 # You are here
+│   └── state-machine.diagram.md  # State machine diagram (generated by visualizer/main.go)
+├── fsm.go                        # Finite State Machine definition (events, states, transitions)
+├── module.go                     # Implementation of the StateMachine module
+└── visualizer
+    └── main.go                   # State machine diagram generator
+```
+
+## High Level Architecture
+
+⚠️ Work in progress ⚠️ - At the time of writing this document, the architecture is still being defined.
+
+This is pretty much a POC at the moment.
+
+The key things to keep in mind are:
+
+- We are using the library "github.com/looplab/fsm" to implement the FSM, essentially here we are wrapping it and integrating it with the rest of the modules. This document won't cover the implementation details of the FSM library which is fairly well documented.
+- The **StateMachineModule** can be accessed via the `bus` from any other `IntegratableModule`
+- State machine transitions will emit `StateMachineTransitionEvent` events that the modules that are interested can listen to.
+- The `node` has a main eventhandler for events that can be used to fan-out the handling to the aforementioned modules that listen to transition events
```

Comment:
```suggestion
High-level implementation details:

- The [github.com/looplab/fsm](https://github.com/looplab/fsm) library is used to implement the FSM
- Pocket builds a wrapper around `looplab/fsm` to integrate with the other modules
- The `StateMachineModule` can be accessed via the `bus` from any other `IntegratableModule`
- State machine transitions emit `StateMachineTransitionEvent` events that subscribed pocket modules can listen to
- The `node` has a central [event handler](../../shared/node.go) for events that fan-out event handling to the relevant modules during state transitions
```

Reply:
I realize these are most nits, but thinking of "what text do I not need to write" really helps with improving documentation